modules/build/src/test/scala/scala/build/tests/BuildOptionsTests.scala

```diff
@@ -29,14 +29,15 @@ class BuildOptionsTests extends munit.FunSuite {
   }
 
   val expectedScalaVersions = Seq(
-    Some("3")      -> defaultScalaVersion,
-    None           -> defaultScalaVersion,
-    Some("2.13")   -> defaultScala213Version,
-    Some("2.12")   -> defaultScala212Version,
-    Some("2")      -> defaultScala213Version,
-    Some("2.13.2") -> "2.13.2",
-    Some("3.0.1")  -> "3.0.1",
-    Some("3.0")    -> "3.0.2"
+    Some("3")         -> defaultScalaVersion,
+    None              -> defaultScalaVersion,
+    Some("2.13")      -> defaultScala213Version,
+    Some("2.12")      -> defaultScala212Version,
+    Some("2")         -> defaultScala213Version,
+    Some("2.13.2")    -> "2.13.2",
+    Some("3.0.1")     -> "3.0.1",
+    Some("3.0")       -> "3.0.2",
+    Some("3.nightly") -> "3.1.3-RC1-bin-20220205-8e2fab7-NIGHTLY"
```

Comment:
wouldn't it break tommorrow when new nightly is released?

Reply:
I think we should just check if in case of `3.nightly` version starts with `3` and ends with `NIGHTLY `